apache-spark-master/Makefile

```diff
@@ -1,24 +1,33 @@
+# Definitions
+spark_versions = 1.5.2 1.6.2 2.0.1
+common_git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../apache-spark-common | cut -f1 -d " ")
+gen_directories = $(spark_versions:=.spark)
 build_tool = runtime-container.DONE
-git_commit ?= $(shell git log --pretty=oneline -n 1 -- ../apache-spark-master | cut -f1 -d " ")
-nametag = quay.io/ucsc_cgl/apache-spark-master:1.5.2--${git_commit}
 
-build: ${build_tool}
+build: $(build_tool)
 
-${build_tool}: Dockerfile run-master.sh
+%.spark: sparkver=$(@:.spark=)
+%.spark: Makefile.template Dockerfile.template
+	mkdir -p $@
+	cp run-master.sh ${@}/
+	sed -e "s:SPARKVER:$(sparkver):g" -e "s:COMMONCOMMIT:$(common_git_commit):g" Dockerfile.template > ${@}/Dockerfile
```

Comment:
I'm somewhat agnostic. I agree that `make` has many idiosyncrasies, but this code is pretty basic. Let me peek back at the `Dockerfile.py` approach and sleep on it and I'll ping back here.